week_4/quiz-trivia-app/src/app/screens/quiz-page/answer/answer.component.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  AfterViewInit,
+  ChangeDetectorRef,
+  Component,
+  Input,
+  OnInit,
+} from '@angular/core';
+import { QuizService } from '../../../data/quiz.service';
+import { Quiz } from '../../../data/quiz.model';
+import { CommonModule } from '@angular/common';
+import { StorageService } from '../../../storage.service';
+
+@Component({
+  selector: 'app-answer',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './answer.component.html',
+  styleUrl: './answer.component.css',
+})
+export class AnswerComponent implements OnInit, AfterViewInit {
+  @Input() subjectObject!: Quiz | undefined;
+
+  num = 0;
+  answers!: string[];
+  currentAnswer = this.subjectObject?.questions[this.num].answer;
+  selectedAnswer: string | null = null;
+  letter: string[] = ['A', 'B', 'C', 'D'];
+  hoverIndex = -1;
+  selectIndex = -1;
+  isCorrect!: boolean;
+  isSubmitted = false;
+  renderWarning = false;
+  correctAnswer!: string;
+  correctAnswerIndex!: number;
+  totalQuestions: number | undefined = undefined;
```

Comment:
dont use undefined as value. you can assign to 0.

Reply:
done